src/controllers/music/songs.js

```diff
@@ -8,197 +7,207 @@ import * as userSettings from '../../scripts/settings/userSettings';
-            return getPageData(tabContent).view;
-        };
+    const self = this;
+    const data = {};
+    let isLoading = false;
 
-        function initPage(tabContent) {
-            tabContent.querySelector('.btnFilter').addEventListener('click', function () {
-                self.showFilterMenu();
+    self.showFilterMenu = function () {
+        import('../../components/filterdialog/filterdialog').then(({default: filterDialogFactory}) => {
+            const filterDialog = new filterDialogFactory({
+                query: getQuery(tabContent),
+                mode: 'songs',
+                serverId: ApiClient.serverId()
             });
-            tabContent.querySelector('.btnSort').addEventListener('click', function (e) {
-                libraryBrowser.showSortMenu({
-                    items: [{
-                        name: globalize.translate('OptionTrackName'),
-                        id: 'Name'
-                    }, {
-                        name: globalize.translate('Album'),
-                        id: 'Album,SortName'
-                    }, {
-                        name: globalize.translate('AlbumArtist'),
-                        id: 'AlbumArtist,Album,SortName'
-                    }, {
-                        name: globalize.translate('Artist'),
-                        id: 'Artist,Album,SortName'
-                    }, {
-                        name: globalize.translate('OptionDateAdded'),
-                        id: 'DateCreated,SortName'
-                    }, {
-                        name: globalize.translate('OptionDatePlayed'),
-                        id: 'DatePlayed,SortName'
-                    }, {
-                        name: globalize.translate('OptionPlayCount'),
-                        id: 'PlayCount,SortName'
-                    }, {
-                        name: globalize.translate('OptionReleaseDate'),
-                        id: 'PremiereDate,AlbumArtist,Album,SortName'
-                    }, {
-                        name: globalize.translate('Runtime'),
-                        id: 'Runtime,AlbumArtist,Album,SortName'
-                    }],
-                    callback: function () {
-                        getQuery(tabContent).StartIndex = 0;
-                        reloadItems(tabContent);
-                    },
-                    query: getQuery(tabContent),
-                    button: e.target
-                });
+            Events.on(filterDialog, 'filterchange', function () {
+                getQuery(tabContent).StartIndex = 0;
+                reloadItems(tabContent);
             });
-        }
-
-        initPage(tabContent);
+            filterDialog.show();
+        });
+    };
+
+    self.getCurrentViewStyle = function () {
+        return getPageData(tabContent).view;
+    };
+
+    function shuffle() {
+        ApiClient.getItem(ApiClient.getCurrentUserId(), params.topParentId).then(function (item) {
+            getQuery();
```

Comment:
Is `getQuery` necessary?
We may need to use a filter on shuffle at some point, but the latter doesn't support filtering yet.